File: java/src/Ui/HelpCommand.java
package Ui;

import java.io.PrintStream;

class HelpCommand implements UserCommand {

	private PrintStream out;

	HelpCommand(PrintStream out) {
		this.out = out;
	}

	public void run(UserInput cmd) {
		out.println("Customer actions:");
		out.println("  a sku [num]             Add SKU to basket; num defaults to 1");
		out.println("  b                       List basket contents");
		out.println("  c                       Checkout");
		out.println("  d sku                   Show detailed product info for SKU");
		out.println("  p                       List all products");
		out.println();
		out.println("Warehouse actions:");
		out.println("  r aisle loc sku num     Replenish SKU with num additional units");
		out.println("  s                       Show stock levels");
		out.println();
		out.println("General:");
		out.println("  h                       Show this help");
		out.println("  q                       Quit");
	}

}

File: csharp/Ui/HelpCommand.cs
namespace Ui {

	using System;

class HelpCommand : UserCommand {

	public void run(UserInput cmd) {
		Console.WriteLine("Customer actions:");
		Console.WriteLine("  a sku [num]             Add SKU to basket; num defaults to 1");
		Console.WriteLine("  b                       List basket contents");
		Console.WriteLine("  c                       Checkout");
		Console.WriteLine("  d sku                   Show detailed product info for SKU");
		Console.WriteLine("  p                       List all products");
		Console.WriteLine();
		Console.WriteLine("Warehouse actions:");
		Console.WriteLine("  r aisle loc sku num     Replenish SKU with num additional units");
		Console.WriteLine("  s                       Show stock levels");
		Console.WriteLine();
		Console.WriteLine("General:");
		Console.WriteLine("  h                       Show this help");
		Console.WriteLine("  q                       Quit");
	}

}
}